base/math.jl

```diff
@@ -146,15 +146,17 @@ sqrt(x::Float32) = box(Float32,sqrt_llvm(unbox(Float32,x)))
 sqrt(x::Real) = sqrt(float(x))
 @vectorize_1arg Number sqrt
 
-hypot(x::Real, y::Real) = hypot(promote(float(x), float(y))...)
-function hypot{T<:AbstractFloat}(x::T, y::T)
+@inline hypot(x::Real, y::Real) = hypot(promote(float(x), float(y))...)
+@inline function hypot{T<:AbstractFloat}(x::T, y::T)
     x = abs(x)
     y = abs(y)
     if x < y
         x, y = y, x
     end
-    if x == 0
```

Comment:
Unfortunately, I've had some problems with this and this particular change was made to improve the LLVM code.